### PR TITLE
ci: update the base os version to make it work

### PR DESCRIPTION
**Pull request by KazuCocoa**
## Proposed changes

https://github.com/appium/appium/runs/8017168074 failed to run the CI by the OS version issue. 

It seems like the purpose was to run tests on Sauce Labs...? https://github.com/appium/appium/pull/12308
@dpgraham Do you remember if we still need this job? If not, I'll disable this job in the Azure

https://dev.azure.com/AppiumCI/Appium%20CI/_build?definitionId=20&_a=summary

## Types of changes

What types of changes does your code introduce to Appium?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](https://github.com/appium/appium/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla.js.foundation/appium/appium)
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: ci-jobs/generate-docs.yml [hunk 1]
```diff
@@ -1,40 +0,0 @@
-# parameters: # No params yet, but may add later
```
Comment (KazuCocoa) [inline ci-jobs/generate-docs.yml:1]: this one is only used in 1.x branch by https://dev.azure.com/AppiumCI/Appium%20CI/_build?definitionId=38&_a=summary , so we can remove this safely from current master

File: ci-jobs/templates/bundle-template.yml [hunk 2]
```diff
@@ -2,7 +2,7 @@
 jobs:
   - job: bundle_and_upload_appium_zip
     pool:
-      vmImage: 'macOS 10.14'
+      vmImage: 'macOS 11'
```
Comment (mykola-mokhnach) [inline ci-jobs/templates/bundle-template.yml:5]: I believe it's macOS-11 (with a dash)
We could also probably just use latest as this script does not have hard dependencies on image component versions?

File: ci-jobs/templates/bundle-template.yml [hunk 3]
```diff
@@ -11,21 +11,21 @@ jobs:
...
+        versionSpec: '16.x'
     - script: npm install
       displayName: Install NPM Dependencies
     - script: npm run build
       displayName: npm run build
     - script: npm prune --production
       displayName: Get rid of dev dependencies
     - script: |
-        curl -o node-darwin.tar.gz https://nodejs.org/dist/v12.16.1/node-v12.16.1-darwin-x64.tar.gz
+        curl -o node-darwin.tar.gz https://nodejs.org/dist/v16.17.0/node-v16.17.0-darwin-x64.tar.gz
```
Comment (mykola-mokhnach) [inline ci-jobs/templates/bundle-template.yml:22]: can we have the node version stored in a variable?

**Comment by KazuCocoa** [comment]
updated

**Review by mykola-mokhnach — APPROVED** [review]

**Pull request merged**